### PR TITLE
feat: migrated from TSLint to ESLint

### PR DESCRIPTION
**Pull request by vazra**
As per https://github.com/palantir/tslint/issues/4534 , TSLint is deprecated 

Coses #36

### CODE REVIEW

**Comment by vazra** [comment]
https://github.com/electron/electron-quick-start-typescript/pull/39/commits/da0d1c84b8a44270b964b0f72bb38cd438f70a7e resolved conflicts with the base branch. 
 cc: @codebytere

**Comment by vazra** [comment]
cc : @vanessayuenn , @malept , @zeke

File: package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
   "author": "GitHub",
   "license": "CC0-1.0",
   "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^3.2.0",
+    "@typescript-eslint/parser": "^3.2.0",
     "electron": "^9.0.4",
     "tslint": "^6.1.2",
```
Comment (VladNeacsu) [inline package.json:26]: Remove `tslint` from the `packages.json` file as it is no longer used

File: package.json [hunk 2]
```diff
@@ -20,6 +20,8 @@
   "author": "GitHub",
   "license": "CC0-1.0",
   "devDependencies": {
```
Comment (VladNeacsu) [inline package.json:22]: You should add `eslint` to the `devDependencies`

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "build": "tsc",
     "watch": "tsc -w",
-    "lint": "tslint -c tslint.json -p tsconfig.json",
+    "lint": "eslint . --ext .ts",
```
Comment (VladNeacsu) [inline package.json:8]: You should pass a `source` folder to the `eslint` command that way you won't need the `.eslintignore` file

Comment (vazra) [reply package.json:8]: removed .eslintignore, @VladNeacsu adding 'src' instead of '.' is fine, right?

File: package.json (hunk 2 above)
Comment (vazra) [reply package.json:22]: fixed it, it was done, but got reverted when merged from master.

File: package.json (hunk 1 above)
Comment (vazra) [reply package.json:26]: fixed it, it was done, but got reverted when merged from master.

**Pull request closed without merging**